Common/src/main/java/net/frankheijden/serverutils/common/tasks/PluginWatcherTask.java

```diff
@@ -65,12 +68,22 @@ public void run() {
                 WatchKey key = watchService.take();
                 for (WatchEvent<?> event : key.pollEvents()) {
                     if (file.getName().equals(event.context().toString())) {
-                        send(WatchResult.CHANGE);
+                        String previousHash = hash;
+                        hash = FileUtils.getHash(file);
+                        if (task != null) {
+                            //noinspection unchecked
+                            taskManager.cancelTask(task);
+                        }
+                        task = ServerUtilsApp.getPlugin().getTaskManager().runTaskLater(() -> {
+                            if (hash.equals(previousHash)) {
+                                send(WatchResult.CHANGE);
 
-                        taskManager.runTask(() -> {
-                            pluginManager.reloadPlugin(pluginName);
-                            file = pluginManager.getPluginFile(pluginName);
-                        });
+                                taskManager.runTask(() -> {
```

Comment:
Since the `runTaskLater` method is synchronous, this extra scheduling could probably be removed ^^